src/components/router/Router.brs

```diff
@@ -51,22 +48,45 @@ sub resetHistory(rootPath = "" as String)
   m._history = []
 
   if (rootPath <> "")
-    m._history.push({ path: rootPath, params: {} })
+    rootRoute = CreateObject("roSGNode", "ActivatedRoute")
+    rootRoute.setFields({ path: rootPath })
```

Comment:
we could add helper function like `_createActivatedRoute(data)` and used here and in `_getPreviousRoute`